loader/db.go

```diff
@@ -104,14 +106,29 @@ func (conn *Conn) executeSQL(ctx *tcontext.Context, queries []string, args ...[]
 		FirstRetryDuration: 2 * time.Second,
 		BackoffStrategy:    retry.LinearIncrease,
 		IsRetryableFn: func(retryTime int, err error) bool {
-			ctx.L().Warn("execute statements", zap.Int("retry", retryTime),
-				zap.String("queries", utils.TruncateInterface(queries, -1)),
-				zap.String("arguments", utils.TruncateInterface(args, -1)),
-				log.ShortError(err))
 			tidbExecutionErrorCounter.WithLabelValues(conn.cfg.Name).Inc()
-			return retry.IsRetryableError(err)
+			if retry.IsConnectionError(err) {
+				err = conn.resetConn(ctx)
```

Comment:
only reset connection for `executeSQL` but not for `querySQL`? same in _syncer/db.go_.

Reply:
can we reset the connection in `BaseConn`?

Reply:
if there is a reset function in `BaseConn`, it need a `*BaseDB` object to generate `BaseConn` and close old `BaseConn`, so which one is better?
1. make `*BaseDB` a member of `BaseConn`, then each `BaseConn` shared one `*BaseDB`
2. make `*BaseDB` a argument of `BaseConn` reset function

Reply:
in `sql.Conn`, it holds `*sql.DB`. can we learn from it (the first one you post)?

Reply:
let me try

Reply:
I moved `BaseDB` as ParentDB in `BaseConn`, and then `BaseConn` can [reset](https://github.com/pingcap/dm/blob/a7f91d8cedf649ad6848a7b2d91d73107043a6f1/pkg/conn/baseconn.go#L81) itself, and add some comment in [baseconn](https://github.com/pingcap/dm/pull/266/files#diff-e25c1def89789065c4904c1d5adc0867R29-R32)